llm_bioinformatics_research/src/Components/ResetPassword/ResetPassword.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import { TextField, Button, Container, Typography, Box, IconButton } from '@mui/material';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import config from '../../config.json';
+import './ResetPassword.css';
+
+function ResetPassword() {
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [passwordVisible, setPasswordVisible] = useState(false);
+  const [confirmPasswordVisible, setConfirmPasswordVisible] = useState(false);
+  const [message, setMessage] = useState('');
+  const [error, setError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const port = config.port;
+  const token = searchParams.get("token");
+
+  const validatePassword = (password) => {
+    if (password.length < 8) return 'Password must be at least 8 characters long.';
+    if (!/[A-Z]/.test(password)) return 'Password must include at least one uppercase letter.';
+    if (!/[a-z]/.test(password)) return 'Password must include at least one lowercase letter.';
+    if (!/\d/.test(password)) return 'Password must include at least one number.';
+    if (!/[@$!%*#?&]/.test(password)) return 'Password must include at least one special character.';
+    return '';
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    
+    // Validate passwords
+    const passwordValidationError = validatePassword(newPassword);
+    setPasswordError(passwordValidationError);
+    if (passwordValidationError) return;
+  
+    if (newPassword !== confirmPassword) {
+      setConfirmPasswordError("Passwords do not match.");
+      return;
+    } else {
+      setConfirmPasswordError('');
+    }
+  
```

Comment:
In addition to the existing validation, I think we should implement a check that ensures that the new password is not the same as the existing password (we don't want the user to change it to the same password for security reasons).